tools/build_env/README.md

```diff
@@ -0,0 +1,75 @@
+# Hermetic Build Environment (Beta)
+
+This directory contains the files needed to create a Docker image that runs a
+build environment for uProxy development without the need to install any tools
+other than Docker. The only dependencies for the hermetic development
+environment is Docker and the uProxy repository.
+
+This allows us to reproduce the build the same way in any machine, and makes it
+easier to set up the development environment for new uProxy developers.
+
+## Usage
+
+You first need to install Docker:
+https://docs.docker.com/engine/installation/.
+
+Then you will be calling `./tools/build_env/build_env.sh` to interact with the
+build environment. You must **call `build_env.sh` from your uProxy repository
+root** (where the .git directory is).
+
+In the first usage, `build_env` will fetch `uproxy/build` Docker image tagged
+with the value from `IMAGE_VERSION.txt`, if it's not yet in your machine. A new
+image will also be fetch as needed whenever `IMAGE_VERSION.txt` is modified.
+
+### Entering the environment
+
+Use the `enter` command:
+````
+./tools/build_env/build_env.sh enter
+````
+
+This will:
+
+1. Start a new Docker container with the `uproxy/build:<tag>` image
```

Comment:
You need a leading space on these items for the markdown renderer to make it a list (Github's preview feature is handy for tweaking this formatting stuff).